### PR TITLE
RadioGroup improvements

### PR DESCRIPTION
**Pull request by dweymouth**
### Description:

Fixes: #3786, #3543 (for RadioGroup)

Improvements to RadioGroup
* Now supports multiple items with the same label, the one tapped will be selected only
* Removed slice of `radioItem` from the widget type (only renderer deals with the radioItems)
* Except for the `*_Duplicate*` tests, no test logic is changed; only other test changes are how we access items now that internal impl is different

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

#### Where applicable:

- [ ] Public APIs match existing style and have Since: line.
- [ ] Any breaking changes have a deprecation path or have been discussed.
- [ ] Check for binary size increases when importing new modules.

### CODE REVIEW

**Comment by Jacalz** [comment]
Nice. Well done :)

**Comment by Jacalz** [comment]
A quick look at the failing tests seem to suggest that disabling the widget no longer works as expected?

**Comment by dweymouth** [comment]
> A quick look at the failing tests seem to suggest that disabling the widget no longer works as expected?

Interesting, I tested the disabling manually and it seems to work fine

**Comment by andydotxyz** [comment]
Ah, got it - the group is not passing the item state into the items when It creates them - only in the `Refresh()` call is it applied.

I was able to fix some tests as follows, the rest would be passed similarly I imagine:
```go
@@ -15,9 +15,11 @@ var _ desktop.Hoverable = (*radioItem)(nil)
 var _ fyne.Tappable = (*radioItem)(nil)
 var _ fyne.Focusable = (*radioItem)(nil)
 
-func newRadioItem(label string, onTap func(*radioItem)) *radioItem {
+func newRadioItem(label, sel string, onTap func(*radioItem)) *radioItem {
        i := &radioItem{Label: label, onTap: onTap}
        i.ExtendBaseWidget(i)
+
+       i.Selected = sel == label
        return i
 }
```

**Comment by dweymouth** [comment]
Hmm, good find, not sure how to fix it though as 

> i.Selected = sel == label

is just reintroducing the same issue (all items with the same label would be selected).

Maybe the tests need to be updated since it works in the "real world" so-to-speak

**Comment by andydotxyz** [comment]
It only works real world because it's refreshing after display for some reason.
Initial state should be correct.
Maybe some of the new code needs to be called on setup?

**Comment by dweymouth** [comment]
Looks like all remaining test failures are flukes/unrelated

**Comment by andydotxyz** [comment]
> Looks like all remaining test failures are flukes/unrelated

Agreed. One was fixed on the pending theme widget branch and the other is intermittent.

File: widget/radio_group_extended_test.go [hunk 1]
```diff
@@ -43,7 +42,8 @@ func TestRadioGroup_Extended_Unselected(t *testing.T) {
 		selected = sel
 	})
 	radio.Selected = selected
-	radio.items[0].Tapped(&fyne.PointEvent{Position: fyne.NewPos(theme.Padding(), theme.Padding())})
+	radio.Refresh() // sets up selectedIndex
```
Comment (andydotxyz) [inline widget/radio_group_extended_test.go:45]: This line should not be needed any more?

File: widget/radio_group_internal_test.go [hunk 2]
```diff
@@ -57,15 +57,16 @@ func TestRadioGroup_Unselected(t *testing.T) {
 		selected = sel
 	})
 	radio.Selected = selected
-	radio.items[0].Tapped(&fyne.PointEvent{Position: fyne.NewPos(theme.Padding(), theme.Padding())})
+	radio.Refresh() // sets up selectedIndex
```
Comment (andydotxyz) [inline widget/radio_group_internal_test.go:60]: Is this line needed? I don't think it should be

**Review by andydotxyz — CHANGES_REQUESTED** [review]
This is looking great - but I think two Refreshes in the tests should not be needed now that the initial state is correct?

**Review by Jacalz — APPROVED** [review]
I'm happy when Andrew is happy. Thanks for fixing this. I'm glad to see that we allocating less garbage for the GC to deal with :)

**Review by andydotxyz — APPROVED** [review]
Perfect thanks. I'll merge this in over the failing tests then cherry pick the fix off my long standing PR to get this resolved too.

**Pull request merged**